fleet/postgres/agents.go

```diff
@@ -391,19 +391,22 @@ func (r agentRepository) RetrieveByID(ctx context.Context, ownerID string, thing
 	q := `SELECT 
 			mf_thing_id, 
 			name, 
-			mf_owner_id, 
+			agents.mf_owner_id, 
 			mf_channel_id, 
 			ts_created, 
 			orb_tags, 
 			agent_tags, 
 			agent_metadata, 
 			state, 
 			last_hb_data, 
-			ts_last_hb 
+			ts_last_hb,
+			json_build_object('groups', json_strip_nulls(json_agg(json_build_object('group_id', agm.agent_groups_id, 'name', agm.agent_groups_name, 'channel', agm.group_mf_channel_id)))) matching_groups
 		FROM agents 
+			LEFT JOIN agent_group_membership agm on agents.mf_owner_id = agm.mf_owner_id and agents.mf_thing_id = agm.agent_mf_thing_id
 		WHERE 
-			mf_thing_id = $1 
-			AND mf_owner_id = $2;`
+		mf_thing_id = $1 
+		AND agents.mf_owner_id = $2
+		group by mf_thing_id, name, agents.mf_owner_id, mf_channel_id, ts_created, orb_tags, agent_tags, agent_metadata, state, last_hb_data, ts_last_hb;`
```

Comment:
The need for this "group by" seems really weird to me - we will have to list all the columns? When a new column is added, we will have to remember to add it here to?

Maybe it makes sense to make this a separate query, rather than a join. In fact, if we go further and make it an entirely separate (new) endpoint, then the currently GET agent is unaffected, and the UI can make a concurrent request, possibly improving efficiency.